nixos/nextcloud: avoid duplicate X-Robots-Tag header, remove option nginx.recommendedHttpHeaders

teutat3s opened the pull request:
We already set this header in `nginx` https://github.com/NixOS/nixpkgs/blob/f8b7dbc46d330984bb52722a4e5df70fd364ab1d/nixos/modules/services/web-apps/nextcloud.nix#L1539 and a duplicate header can lead to issues, like: https://github.com/nextcloud/notes-android/issues/2848.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

provokateurin commented:
I would do it the other way around and not set the header in nginx, if it  is already set by Nextcloud itself.
In any case, multiple X-Robots-Tag are fine and should not cause any issues (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/X-Robots-Tag for some examples). I feel like this is up to the app to handle correctly and not the server deployment.

teutat3s commented:
If we don't set it in `nginx`, static files served by `nginx` won't have the header.

SuperSandro2000 commented:
Does fastcgi has functionallity like more_set_headers that we can avoid duplicating the header?

SuperSandro2000 reviewed (approved):
Since we follow the upstream htaccess file with that.

provokateurin commented on nixos/modules/services/web-apps/nextcloud.nix:
```diff
@@ -1536,7 +1536,6 @@ in
             index index.php index.html /index.php$request_uri;
             ${lib.optionalString (cfg.nginx.recommendedHttpHeaders) ''
               add_header X-Content-Type-Options nosniff;
-              add_header X-Robots-Tag "noindex, nofollow";
```
It was moved further down and out of the condition.

teutat3s committed:
nixos/nextcloud: avoid duplicate X-Robots-Tag header

We already set this header in nginx, and a duplicate header can lead to
issues, like: https://github.com/nextcloud/notes-android/issues/2848

teutat3s committed:
nixos/nextcloud: recommended nginx headers should

not be optional, because upstream recommends them
https://docs.nextcloud.com/server/stable/admin_manual/installation/harden_server.html#serve-security-related-headers-by-the-web-server
